trainer/lib/trainer/xcresult.rb

```diff
@@ -184,16 +184,12 @@ def find_failure(failures)
           #     or identifier: "TestThisDude/testFailureJosh2" (when Objective-C)
 
           found_failure = failures.find do |failure|
-            # Clean test_case_name to match identifier format
-            # Sanitize for Swift by replacing "." for "/"
-            # Sanitize for Objective-C by removing "-", "[", "]", and replacing " " for ?/
-            sanitized_test_case_name = failure.test_case_name
-                                              .tr(".", "/")
-                                              .tr("-", "")
-                                              .tr("[", "")
-                                              .tr("]", "")
-                                              .tr(" ", "/")
```

Comment:
In my case, this ended up being the unmet expectation - not all spaces were replaced by `/` in the `identifier`. So the comparison ended up being [somewhat redacted due to proprietary info]:

```
[1] pry(#<Trainer::XCResult::ActionTestMetadata>)> self.identifier
=> "MyPrivateSpec/my test condition, should return an error()"
[2] pry(#<Trainer::XCResult::ActionTestMetadata>)> sanitized_test_case_name
=> "MyPrivateSpec/my/test/condition,/should/return/an/error()"
```

Anyways, the bet I'm making with this PR is: rather than have any expectation of how `self.identifier` is formatted, sanitize both it and `failure.test_case_name` in the same fashion then compare.